### PR TITLE
Implemented the load more functionality. 

### PR DESCRIPTION
**Pull request by ArtheeswariR**

### CODE REVIEW

File: addon/components/x-tree-branch.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+
+export default class XTreeBranchComponent extends Component {
+  get loadMoreText() {
```
Comment (kumkanillam) [inline addon/components/x-tree-branch.js]: Are we using `loadMoreText`?

File: addon/components/x-tree-branch.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+	if(this.args.model.length !=0){
+		return this.args.model[0].loadMoreText;
+	}
+  }
+   
+  @action
+   loadingNext(modell){
```
Comment (kumkanillam) [inline addon/components/x-tree-branch.js]: we can avoid defining `loadingNext` action in this component js file. Reason is, we are are writting anything special in it.
Have same method name onLoad means always onLoad.

**Review by kumkanillam — CHANGES_REQUESTED** [review]

File: addon/components/x-tree-branch.hbs [hunk 3]
```diff
@@ -41,6 +41,6 @@
     {{/if}}
   {{/each}}
 {{#if (eq @model.lastObject.hasNextPage true)}}
-	<span class="load-more" {{on "click" (fn this.loadingNext @model.lastObject.parentObject)}} >+{{@model.lastObject.parentObject.loadingTriggerCount}} more </span>
+	<span class="x-tree-load-more" {{on "click" (fn this.onLoad @model.lastObject.parentObject)}} >+{{@model.lastObject.parentObject.loadingTriggerCount}} {{@model.lastObject.parentObject.loadMoreText}} </span>
```
Comment (nagarajanpp8) [inline addon/components/x-tree-branch.hbs]: loadMoreText => loadMoreLabel

File: addon/components/x-tree-branch.hbs [hunk 4]
```diff
@@ -41,6 +41,6 @@
     {{/if}}
   {{/each}}
 {{#if (eq @model.lastObject.hasNextPage true)}}
```
Comment (nagarajanpp8) [inline addon/components/x-tree-branch.hbs]: No need to have a eq comparison for true value

**Review by nagarajanpp8 — APPROVED** [review]

File: addon/components/x-tree-branch.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+
+export default class XTreeBranchComponent extends Component {
+  
+  @action
+   onLoad(modell){
```
Comment (kumkanillam) [inline addon/components/x-tree-branch.js]: - modell to model.
- Can we directly use the onLoad function in the component, as we are not writing anything inside the method.

**Review by kumkanillam — CHANGES_REQUESTED** [review]

File: addon/components/x-tree-branch.js (hunk 5 above)
Comment (Rtheeswari) [reply addon/components/x-tree-branch.js:7]: I think we cant use it directly. I referred the addon code, they described other properties such as onSelect, onHover, onHoverout... like, 
` @action
  mouseEnter() {
    if (!get(this.args.model, 'isDisabled')) {
      set(this.args.model, 'isSelected', true);
    }


    if (this.args.onHover) {
      this.args.onHover(this.args.model);
    }
  }

  @action
  mouseLeave() {
    set(this.args.model, 'isSelected', false);

    if (this.args.onHoverOut) {
      this.args.onHoverOut(this.args.model);
    }
  }`

**Review by kumkanillam — APPROVED** [review]

**Pull request merged**